rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import json
+
+from prisma.models import AgentGraph, AgentNode, AgentNodeLink, AgentNodeExecution
+
+from autogpt_server.data.db import BaseDbModel
+
+
+class Edge(BaseDbModel):
+    input_node: str
+    input_name: str
+    output_node: str
+    output_name: str
+
+    @staticmethod
+    def from_db(edge: AgentNodeLink):
+        return Edge(
+            id=edge.id,
+            input_node=edge.agentNodeInputId,
+            input_name=edge.inputName,
+            output_node=edge.agentNodeOutputId,
+            output_name=edge.outputName,
+        )
+
+
+class Node(BaseDbModel):
+    block_name: str
+    input_default: dict[str, str] = {}  # dict[input_name, input_value]
+    input_schema: dict[str, str] = {}  # dict[input_name, type]
+    output_schema: dict[str, str] = {}  # dict[output_name, type]
+    input_nodes: dict[str, str] = {}  # dict[input_name, node_id]
+    output_nodes: dict[str, str] = {}  # dict[output_name, node_id]
```

Comment:
Output node name/id is both in `Node` and in `Edge` which means there are multiple sources of truth and this is error prone (e.g. it changes in one place but not in the other). There should be only one place where the output `Node` is set, maybe in `Edge` because it's responsible to specify connection. If there's a need to have it in `Node` as well this should be resolved dynamically.

Reply:
Removed the edgelist